Sources/Sentry/Profiling/SentryCaptureTransactionWithProfile.mm

```diff
@@ -0,0 +1,50 @@
+#import "SentryCaptureTransactionWithProfile.h"
+
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+
+#    import "SentryDispatchQueueWrapper.h"
+#    import "SentryHub+Private.h"
+#    import "SentryLog.h"
+#    import "SentryProfiledTracerConcurrency.h"
+#    import "SentryProfiler+Private.h"
+#    import "SentryProfilerSerialization.h"
+#    import "SentryProfilerState.h"
+#    import "SentrySwift.h"
+#    import "SentryTracer+Private.h"
+#    import "SentryTransaction.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+void
+sentry_captureTransactionWithProfile(SentryHub *hub, SentryDispatchQueueWrapper *dispatchQueue,
+    SentryTransaction *transaction, NSDate *startTimestamp)
+{
+    const auto profiler = sentry_profilerForFinishedTracer(transaction.trace.internalID);
+    if (!profiler) {
+        [hub captureTransaction:transaction withScope:hub.scope];
+        return;
+    }
+
+    const auto profilingData = [profiler.state copyProfilingData];
```

Comment:
@armcknight, when I move this into the async block the profiles the SDK shortens the profiles uploaded to Sentry, but I can't replicate this behavior in the `testCaptureTransactionWithProfile_SetsCorrectAmountOfSamples`. I would like to have a failing test for this to ensure we don't break this by mistake. Maybe the tests with the mocks behave differently than the actual profiling code, or I just don't get it.

Reply:
OK, then it was a false alarm. I noticed that some app start transactions with profiles also sometimes have a shorter profile on the main branch: see [profile](https://sentry-sdks.sentry.io/profiling/profile/sentry-cocoa/2613a9cccfad44feaf8faccff7afb90c/flamegraph/?colorCoding=by%20library&query=&sorting=call%20order&tid=259&view=top%20down)
![CleanShot 2024-10-09 at 10 43 19@2x](https://github.com/user-attachments/assets/51541f57-0433-4086-8959-95c552527861)

I guess that's a different problem then. Are you aware of such a problem, @armcknight?